charts/seed-controlplane/charts/etcd/templates/etcd-statefulset.yaml

```diff
@@ -59,50 +59,108 @@ spec:
   selector:
     matchLabels:
       app: etcd-statefulset
-      role: {{.Values.role}}
+      role: {{ .Values.role }}
   template:
     metadata:
       labels:
         app: etcd-statefulset
-        role: {{.Values.role}}
+        role: {{ .Values.role }}
     spec:
       containers:
-      - name: etcd-container
+      - name: etcd
         image: {{ index .Values.images "etcd" }}
         imagePullPolicy: IfNotPresent
         command:
-        - "/usr/local/bin/etcd"
-        - "--data-dir=/var/etcd/data"
-        - "--name=etcd-{{.Values.role}}"
-        - "--advertise-client-urls=http://etcd-{{.Values.role}}-0.etcd-{{.Values.role}}:2379"
-        - "--listen-peer-urls=http://0.0.0.0:2380"
-        - "--listen-client-urls=http://0.0.0.0:2379"
-        - "--initial-advertise-peer-urls=http://etcd-{{.Values.role}}-0.etcd-{{.Values.role}}:2380"
-        - "--initial-cluster=etcd-{{.Values.role}}=http://etcd-{{.Values.role}}-0.etcd-{{.Values.role}}:2380"
-        - "--initial-cluster-state=new"
-        - "--initial-cluster-token=new"
+        - /bin/sh
+        - -ec
+        - 'while true; do
+wget http://localhost:8080/initialization/status -S -O status;
+STATUS=`cat status`;
+case $STATUS in
+  "New")        wget http://localhost:8080/initialization/start -S -O - ;;
+  "Progress")   sleep 1;
+                continue;;
+  "Failed")     continue;;
+  "Successful") exec etcd --data-dir=/var/etcd/data
+                --name=etcd-{{ .Values.role }}
+                --advertise-client-urls=http://0.0.0.0:2379
+                --listen-peer-urls=http://0.0.0.0:2380
+                --listen-client-urls=http://0.0.0.0:2379
+                --initial-advertise-peer-urls=http://0.0.0.0:2380
+                --initial-cluster=etcd-{{ .Values.role }}=http://0.0.0.0:2380
+                --initial-cluster-state=new
+                --initial-cluster-token=new;;
+esac;
+done'
         livenessProbe:
+          httpGet:
+            path: /metrics
+            port: 8080
+          initialDelaySeconds: 15
+          periodSeconds: 15
+        readinessProbe:
           exec:
             command:
-            - "/bin/sh"
-            - "-ec"
-            - "ETCDCTL_API=3 etcdctl get foo"
-          initialDelaySeconds: 15
-          timeoutSeconds: 15
+            - /bin/sh
+            - -ec
+            - ETCDCTL_API=3 etcdctl get foo
+          initialDelaySeconds: 5
+          periodSeconds: 5
         ports:
-        - name: serverport
-          containerPort: 2380
-        - name: clientport
-          containerPort: 2379
+        - containerPort: 2380
+          name: serverport
+          protocol: TCP
+        - containerPort: 2379
+          name: clientport
+          protocol: TCP
         resources:
           requests:
             cpu: 200m
         volumeMounts:
-        - name: etcd-{{.Values.role}}
-          mountPath: /var/etcd/data
+        - name: etcd-{{ .Values.role }}
+          mountPath: /var/etcd
```

Comment:
Shouldn't that be `/var/etcd/data`?

Reply:
If you check line 128, etcd uses data-dir `/var/etcd/data`, so it doesnt matter. We can keep it /var/etcd/data for keeping it same.